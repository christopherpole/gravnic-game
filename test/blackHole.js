import {
  MOVE_RIGHT,
  MOVE_LEFT,
  ENTITIES,
  changeGravityDirection,
  levelIsComplete,
} from '../src/index';

describe('Black hole entity', () => {
  it('Should suck in any movable entities', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 1 },
        },
        { staticEntity: { id: 3, entityId: ENTITIES.BLACK_HOLE }, movableEntity: null },
      ],
    ];

    const gameStates = changeGravityDirection(gameState, MOVE_RIGHT);

    expect(gameStates).toMatchSnapshot();
    expect(levelIsComplete(gameStates[gameStates.length - 1])).toBe(true);
  });

  it('Should disappear after fading an entity', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 1 },
        },
        { staticEntity: { id: 3, entityId: ENTITIES.BLACK_HOLE }, movableEntity: null },
        {
          staticEntity: { id: 5, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.GLASS, id: 4 },
        },
        {
          staticEntity: { id: 7, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 6 },
        },
      ],
    ];

    const gameStates = changeGravityDirection(gameState, MOVE_LEFT);

    expect(gameStates).toMatchSnapshot();
    expect(levelIsComplete(gameStates[gameStates.length - 1])).toBe(true);
  });
});
