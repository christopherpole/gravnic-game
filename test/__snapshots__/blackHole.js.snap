// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Black hole entity Should disappear after fading an entity 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": Object {
          "entityId": "GLASS",
          "id": 4,
          "shrinking": true,
        },
        "staticEntity": Object {
          "entityId": "BLACK_HOLE",
          "id": 3,
          "shrinking": true,
        },
      },
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "fading": true,
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "fading": true,
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
  ],
]
`;

exports[`Black hole entity Should suck in any movable entities 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 1,
          "shrinking": true,
        },
        "staticEntity": Object {
          "entityId": "BLACK_HOLE",
          "id": 3,
          "shrinking": true,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
  ],
]
`;
